Source/WebCore/rendering/mathml/RenderMathMLBlock.cpp

```diff
@@ -275,6 +275,16 @@ void RenderMathMLBlock::styleDidChange(StyleDifference diff, const RenderStyle*
         setNeedsLayoutAndPrefWidthsRecalc();
 }
 
+void RenderMathMLBlock::insertPositionedChildrenInContainingBlock()
+{
+    for (auto* child = firstChildBox(); child; child = child->nextSiblingBox()) {
+        if (!child->isInFlow()) {
```

Comment:
mmh, I believe `child->isOutOfFlowPositioned()` from your previous patches was correct.  The name `insertPositionedObject` really seems it expects oof positioned, not floats ; and the layout of floats is probably handled differently anyway...

But still if you keep `child->isOutOfFlowPositioned()` here and only layout the in-flow children in `layoutBlock` methods, I believe floats won't be laid out,  so `not-floating-001.html` will crash with an assertion failure. You can see that in your previous patches: https://ews-build.s3-us-west-2.amazonaws.com/macOS-Sonoma-Debug-WK2-Tests-EWS/c1f65df6-11348/imported/w3c/web-platform-tests/mathml/relations/css-styling/floats/not-floating-001-crash-log.txt

In https://w3c.github.io/mathml-core/#css-styling, MathML core contains the same text as in CSS grid & flexbox specs:

> The [float](https://www.w3.org/TR/CSS22/#propdef-float) property does not create floating of elements whose parent's computed display value is block math or inline math, and does not take them out-of-flow.

So I believe the right way to make not-floating-001.html pass is to replicate whatever was done for grid/flexbox.

Reply:
As I see, WebKit currently forces the computed value of float to none for flex and grid displays:

https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/Source/WebCore/style/StyleAdjuster.cpp#492

(This behavior was changed in Chromium [here](https://chromium-review.googlesource.com/c/chromium/src/+/2157889) so that the computed value remains unchanged:  and the behavior extended to `display: math` [here](https://chromium-review.googlesource.com/c/chromium/src/+/4055563))

In order to fix the crash with floats not laid out, I believe you can just do the same for MathML. However, implementing https://bugs.webkit.org/show_bug.cgi?id=278533 is probably too much and I'm not sure it's ok to access the m_element's parent in the style adjuster.

So I wonder whether we could just have a function that performs dummy layout for floats (with some FIXME comment).

Reply:
I see, thanks -- does the current change work?

Reply:
Right, I was still building it sorry! The change is there now